src/pymovements/utils/archives.py

```diff
@@ -88,6 +91,22 @@ def extract_archive(
     if remove_finished:
         source_path.unlink()
 
+    if remove_top_level:
+        children = [str(file) for file in destination_path.glob('*')]
+        # Check if top-level directory has a single child
+        if len(children) == 1:
+            single_child = children[0]
+            if Path.is_dir(Path(single_child)):
+                shutil.copytree(single_child, destination_path, dirs_exist_ok=True)
+                shutil.rmtree(single_child)
+        # Check if top-level directory has just the two children archive and extracted archive
+        elif len(children) == 2 and destination_path == source_path.parent:
```

Comment:
this branch doesn't seem to be covered by tests.
maybe it makes sense to combine the `elif` condition into the `if` a few lines before?